src/SignalRServiceExtension/TriggerBindings/ServerlessHub.cs

```diff
@@ -66,13 +70,35 @@ protected ServerlessHub(IServiceHubContext hubContext = null, IServiceManager se
         /// <summary>
         /// Return a <see cref="SignalRConnectionInfo"/> to finish a client negotiation.
         /// </summary>
-        protected SignalRConnectionInfo Negotiate(string userId = null, IList<Claim> claims = null, TimeSpan? lifeTime = null)
+        [Obsolete("Please use async version instead.")]
+        protected SignalRConnectionInfo Negotiate(string userId = null, IList<Claim> claims = null, TimeSpan? lifeTime = null, HttpContext httpContext = null)
         {
-            return new SignalRConnectionInfo
+            return NegotiateAsync(userId, claims, lifeTime, httpContext).Result;
+        }
+
+        /// <summary>
+        /// Return a <see cref="SignalRConnectionInfo"/> to finish a client negotiation.
+        /// </summary>
+        protected async Task<SignalRConnectionInfo> NegotiateAsync(string userId = null, IList<Claim> claims = null, TimeSpan? lifeTime = null, HttpContext httpContext = null)
+        {
+            if (_hubContext != null)
```

Comment:
> if (_hubContext != null) [](start = 11, length = 25)

why async method need this check while non-async one doesn't?

Reply:
The sync method calls the async method.